packages/dcos-checks/extra/dcos-checks/cmd/binaries_check.go

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2017 NAME HERE <EMAIL ADDRESS>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"github.com/spf13/cobra"
+	"os/exec"
+)
+
+// binariesCmd represents the binaries command
+var binariesCmd = &cobra.Command{
+	Use:   "binaries",
+	Short: "Check for binaries required to install DC/OS",
+	Long: `Check for existence of the following binaries: 
+curl
+wget
+tar
+git
+xz
+unzip
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		RunCheck(NewBinariesCheck("DC/OS binaries/tools health check"))
+	},
+}
+
+func init() {
+	RootCmd.AddCommand(binariesCmd)
+}
+
+// BinariesCheck validates we have the required binaries to install/run DC/OS
+type BinariesCheck struct {
+	Name string
+}
+
+// ID returns a unique check identifier.
+func (c *BinariesCheck) ID() string {
+	return c.Name
+}
+
+// Run the binary check
+func (c *BinariesCheck) Run(ctx context.Context, cfg *CLIConfigFlags) (string, int, error) {
+	err := c.binaryExists(cfg)
+	if err != nil {
+		return "", statusFailure, err
+	}
+	return "", statusOK, nil
+}
+
+func (c *BinariesCheck) binaryExists(cfg *CLIConfigFlags) error {
+	cmds := [...]string{"curl", "wget", "tar", "git", "xz", "unzip"}
+	for _, binary := range cmds {
+		if err := exec.Command("which", binary).Run(); err != nil {
```

Comment:
i am also a little bit concerned around shelling out and running `which` command, probably would be better to get PATH and run `os.stat`, this way you will be able to check for owner and permissions R-X